src/libraries/System.Linq.Expressions/src/System/Linq/Expressions/ExpressionVisitor.cs

```diff
@@ -194,7 +197,7 @@ protected internal virtual Expression VisitBinary(BinaryExpression node)
                 node,
                 node.Update(
                     Visit(node.Left),
-                    VisitAndConvert(node.Conversion, nameof(VisitBinary)),
+                    VisitAndConvert(node.Conversion!, nameof(VisitBinary)),
```

Comment:
>! [](start = 51, length = 1)

`!` shouldn't be necessary if `VisitAndConvert` returns `T?`.

Reply:
right